protocol/communities/manager.go

```diff
@@ -1609,16 +1869,12 @@ func (c *CheckPermissionToJoinResponse) calculateSatisfied() {
 
 	c.Satisfied = false
 	for _, p := range c.Permissions {
-		satisfied := true
 		for _, criteria := range p.Criteria {
-			if !criteria {
-				satisfied = false
-				break
+			if criteria {
+				c.Satisfied = true
+				return
 			}
 		}
-		if satisfied {
-			c.Satisfied = true
-		}
```

Comment:
These changes above do not look right to me.
In fact I've fixed them recently in https://github.com/status-im/status-go/pull/3494#discussion_r1220134354

**All** criteria must be satisfied for a permission to be satisfied.
However, only **one** permission must be satisfied to pass the check.